src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -199,10 +200,15 @@ private ValueTask<HttpConnection> GetConnectionAsync(HttpRequestMessage request,
 
         private async ValueTask<HttpConnection> CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            Stream stream = await ConnectHelper.ConnectAsync(_key, cancellationToken).ConfigureAwait(false);
+            Stream stream =
+                _proxyUri != null ?
+                    (_sslOptions != null ?
+                        await EstablishProxyTunnel(cancellationToken) :
+                        await ConnectHelper.ConnectAsync(_proxyUri.IdnHost, _proxyUri.Port, cancellationToken)) :
+                    await ConnectHelper.ConnectAsync(_host, _port, cancellationToken);
```

Comment:
Rather than await on each of these (and then you'd need ConfigureAwait on each of these as well), the resulting codegen will be better if you just get back the `ValueTask<Stream>` from each and then do a single `await`.  The branching could IMHO also be made easier to understand by inverting the polarity of the checks, e.g.
```C#
ValueTask<Stream> streamTask =
    _proxyUri == null ? ConnectHelper.ConnectAsync(_host, _port, cancellationToken) :
    _sslOptions == null ? ConnectHelper.ConnectAsync(_proxyUri.IdnHost, _proxyUri.Port, cancellationToken) :
    EstablishProxyTunnel(cancellationToken);
Stream stream = await streamTask.ConfigureAwait(false);
```